python/helpers.py

```diff
@@ -0,0 +1,172 @@
+
+
+class ClientBuilder:
+    def __init__(self,
+                 kms_providers=None,
+                 key_alt_name="demo-data-key",
+                 key_db="encryption",
+                 key_coll="__keyVault",
+                 schema=None,
+                 connection_string="mongodb://localhost:27017",
+                 mongocryptd_bypass_spawn=False,
+                 mongocryptd_spawn_path="mongocryptd"):
+
+        super().__init__()
+        if kms_providers is None:
+            raise ValueError("kms_provider is required")
+        self.kms_providers = kms_providers
+        self.key_alt_name = key_alt_name
+        self.key_db = key_db
+        self.key_coll = key_coll
+        self.key_vault_namespace = f"{key_db}.{key_coll}"
+        self.schema = schema
+        self.connection_string = connection_string
+        self.mongocryptd_bypass_spawn = mongocryptd_bypass_spawn
+        self.mongocryptd_spawn_path = mongocryptd_spawn_path
+
+        fle_opts = AutoEncryptionOpts(
+            self.kms_providers,
+            self.key_vault_namespace,
+            mongocryptd_bypass_spawn=self.mongocryptd_bypass_spawn,
+            mongocryptd_spawn_path=self.mongocryptd_spawn_path)
+
+        self.encrypted_client = MongoClient(
+            self.connection_string, auto_encryption_opts=fle_opts)
+
+        self.regular_client = MongoClient(self.connection_string)
+
+        self.key_vault = self.encrypted_client.get_database(
+            self.key_db).get_collection(self.key_coll)
+
+        # clients are required to create a unique sparse index on keyAltNames
+        self.key_vault.create_index("keyAltNames",
+                                    unique=True,
+                                    partialFilterExpression={
+                                        "keyAltNames": {
+                                            "$exists": True
+                                        }
+                                    })
+
+    def find_or_create_data_key(self):
+        """
+        In the guide, we create the data key and then show that it is created by
+        using a find_one query. Here, in implementation, we only create the key if
+        it doesn't already exist, ensuring we only have one local data key.
+
+        We also use the key_alt_names field and provide a key alt name to aid in
+        finding the key in the clients.py script.
+        """
+
+        data_key = self.key_vault.find_one(
+            {"keyAltNames": {"$in": ["demo-data-key"]}})
+
+        if data_key is None:
+            with ClientEncryption(self.kms_providers,
+                                  self.key_vault_namespace,
+                                  self.encrypted_client,
+                                  CodecOptions(uuid_representation=STANDARD)
+                                  ) as client_encryption:
+
+                data_key = client_encryption.create_data_key(
+                    "local", key_alt_names=["demo-data-key"])
```

Comment:
Should you use `self.key_alt_name` here instead of specifying the value a second time?